tests/foreman/cli/test_puppet.py

```diff
@@ -0,0 +1,160 @@
+"""End to end test for Puppet funcionality
+
+@Requirement: Puppet
+
+@CaseAutomation: Automated
+
+@CaseLevel: Acceptance
+
+@CaseComponent: CLI
+
+@TestType: Functional
+
+@CaseImportance: High
+
+@Upstream: No
+"""
+
+from robottelo.config import settings
+from robottelo.decorators import (
+    run_in_one_thread,
+    run_only_on,
+    skip_if_not_set,
+    stubbed,
+    tier3,
+)
+from robottelo.test import CLITestCase
+
+
+@run_in_one_thread
+@skip_if_not_set('clients')
+class PuppetTestCase(CLITestCase):
+    """Implements Puppet test scenario"""
+
+    @classmethod
+    def setUpClass(cls):
+        super(PuppetTestCase, cls).setUpClass()
+        cls.sat6_hostname = settings.server.hostname
+
+    @run_only_on('sat')
+    @stubbed
+    @tier3
+    def test_positive_puppet_scenario(self):
+        """
+        @id: d4fdba9f-6333-4d47-987b-ce920da20d77
+
+        @Steps:
+
+        1. Create an organization and upload a cloned manifest for it.
+
+        2. Enable respective Satellite Tools repos and sync them.
+
+        3. Create a product and a lifecycle environment
+
+        4. Create a puppet repository within the product
+
+        5. Upload motd puppet module into the repository
+        (yet better use parameterizable puppet module and create smart param)
+
+        6. Create a content view
+
+        7. Add Tools repo in the content view
+
+        8. Add the puppet module to the new content view
```

Comment:
Step 6, 7, 8 can be squeezed like: `Create a CV with tools and motd puppet module.`